packages/@aws-cdk/aws-backup/lib/vault.ts

```diff
@@ -0,0 +1,202 @@
+   * @attribute
+   */
+  readonly encryptionKey?: kms.IKey;
+
+  /**
+   * Optional access policy document.
+   * @attribute
+   */
+  readonly accessPolicy?: iam.PolicyDocument;
+}
+
+/**
+ * Represents a Backup Vault.
+ * Backups are stored within this vault.
+ *  new Vault(this, 'MyVault', { props });
+ */
+abstract class BackupVaultBase extends Resource implements IBackupVault {
+  public abstract readonly backupVaultName: string;
+  public abstract readonly backupVaultArn: string;
+  public abstract readonly encryptionKey?: kms.IKey;
+  public abstract readonly accessPolicy?: iam.PolicyDocument;
+}
+
+/**
+ * AWS Backup service can integrate with an SNS Topic and subscribers can
+ * subscribe to such.
+ * The available event types that are made available are as follow.
+ * @see https://docs.aws.amazon.com/aws-backup/latest/devguide/sns-notifications.html
+ */
+export enum EventType {
+  /**
+   * Event for when a backup job has started.
+   */
+  BACKUP_JOB_STARTED = 'BACKUP_JOB_STARTED',
+  /**
+   * Event for when a backup job has been completed.
+   * This event contains successful and failed backup jobs.
+   */
+  BACKUP_JOB_COMPLETED = 'BACKUP_JOB_COMPLETED',
+  /**
+   * Event for when a backup plan has been created.
+   */
+  BACKUP_PLAN_CREATED = 'BACKUP_PLAN_CREATED',
+  /**
+   * Event for when a backup plan has been modified.
+   */
+  BACKUP_PLAN_MODIFIED = 'BACKUP_PLAN_MODIFIED',
+  /**
+   * Event for when a restore job has been started.
+   */
+  RESTORE_JOB_STARTED = 'RESTORE_JOB_STARTED',
+  /**
+   * Event for when the restore job has completed.
+   * This event contains successful and failed restore jobs.
+   */
+  RESTORE_JOB_COMPLETED = 'RESTORE_JOB_COMPLETED',
+  /**
+   * Event for when the recovery point has been modified.
+   */
+  RECOVERY_POINT_MODIFIED = 'RECOVERY_POINT_MODIFIED',
+}
+
+/**
+ * Represents the configuration of a vault's notification.
+ */
+export interface VaultNotifications {
+  /**
+   * An SNS topic in which AWS Backup service will publish all the events.
+   */
+  readonly topic: sns.ITopic;
+  /**
+   * All the event that will get published into the topic.
+   */
+  readonly events: EventType[];
+}
+
+/**
+ * Represents the backup vault's properties.
+ */
+export interface BackupVaultProps {
+  /**
+   * The name to assign the vault.
+   * The name is case-sensitive and must be between 2 and 50 characters or hyphens.
+   * @see https://docs.aws.amazon.com/aws-backup/latest/devguide/creating-a-vault.html
+   */
+  readonly backupVaultName: string;
+  /**
+   * Optional KMS Key to encrypt the backups.
+   * @see https://docs.aws.amazon.com/aws-backup/latest/devguide/creating-a-vault.html
+   * @default Backup Service will create one with the alias `aws/backup`.
+   */
+  readonly encryptionKey?: kms.IKey;
+  /**
+   * Notification configuration based on SNS Topics.
+   * @default configures no notifications
+   */
+  readonly notifications?: VaultNotifications;
```

Comment:
I would not nest the props here, how about `notificationTopic` and `notificationEvents` as top level props and a default of all events if `notificationTopic` is defined but `notificationsEvents` is not.

(it's not because the L1 nests its props that the L2 should also do it)